core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -256,52 +256,47 @@ protected void doStop() throws Exception {
       server.addConnector(connector);
     }
 
-    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
-    context.setContextPath("/");
-
+    // The configuration for the JAX-RS REST service
+    ResourceConfig resourceConfig = new ResourceConfig();
 
-    ServletHolder defaultHolder = new ServletHolder("default", DefaultServlet.class);
-    defaultHolder.setInitParameter("dirAllowed", "false");
+    configureResourceExtensions(resourceConfig);
+    configureBaseApplication(resourceConfig, combinedMetricsTags);
 
-    ResourceCollection staticResources = getStaticResources();
-    if (staticResources != null) {
-      context.setBaseResource(staticResources);
+    // Support mono context setup
+    if (appCtx.size() == 0) {
+      appCtx.add(new DefaultContext(resourceConfig));
     }
 
-    configureSecurityHandler(context);
+    HandlerCollection handlers = new HandlerCollection();
+    for (ApplicationContext ctx : appCtx) {
 
-    if (isCorsEnabled()) {
-      String allowedOrigins = config.getString(RestConfig.ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG);
-      FilterHolder filterHolder = new FilterHolder(CrossOriginFilter.class);
-      filterHolder.setName("cross-origin");
-      filterHolder.setInitParameter(
-          CrossOriginFilter.ALLOWED_ORIGINS_PARAM, allowedOrigins
+      // Isolate contexts by making a defensive copy of the application config
+      ResourceConfig ctxConfig = ctx.ctxConfig;
```

Comment:
1. I can't see the copy being made.
2. What is the difference between config and ctxConfig? Things are getting pretty confusing for me here when I read it.

Reply:
That actually got backed out after I decided to give each context its own ResourceConfig. 

Yes `config` is the application config (RestConfig) ctxConfig is the ResourceConfig (Jersey)